src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -3,102 +3,197 @@
+            a.getAssignmentId(),
+            a.getTitle(),
+            a.getDueDate().toString(),
+            a.getSection().getCourse().getCourseId(),
+            a.getSection().getSecId(),
+            a.getSection().getSectionNo()
+        );
     }
 
     // delete assignment for a section
-    // logged in user must be instructor of the section
+    // logged-in user must be instructor of the section
     @DeleteMapping("/assignments/{assignmentId}")
     public void deleteAssignment(@PathVariable("assignmentId") int assignmentId) {
 
-        // TODO
+        Assignment a = assignmentRepository.findByAssignmentId(assignmentId);
+        assignmentRepository.delete(a);
     }
 
     // instructor gets grades for assignment ordered by student name
     // user must be instructor for the section
     @GetMapping("/assignments/{assignmentId}/grades")
     public List<GradeDTO> getAssignmentGrades(@PathVariable("assignmentId") int assignmentId) {
 
-        // TODO remove the following line when done
-
         // get the list of enrollments for the section related to this assignment.
-		// hint: use te enrollment repository method findEnrollmentsBySectionOrderByStudentName.
+        // hint: use te enrollment repository method
+        // findEnrollmentsBySectionOrderByStudentName.
         // for each enrollment, get the grade related to the assignment and enrollment
-		//   hint: use the gradeRepository findByEnrollmentIdAndAssignmentId method.
-        //   if the grade does not exist, create a grade entity and set the score to NULL
-        //   and then save the new entity
-
-        return null;
+        // hint: use the gradeRepository findByEnrollmentIdAndAssignmentId method.
+        // if the grade does not exist, create a grade entity and set the score to NULL
+        // and then save the new entity
+
+        Assignment a = assignmentRepository.findByAssignmentId(assignmentId);
+
+        if (a == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Assignment " + assignmentId + " not found");
+        }
+
+        List<Enrollment> enrollments = enrollmentRepository
+                .findEnrollmentsBySectionNoOrderByStudentName(a.getSection().getSectionNo());
+
+        List<GradeDTO> grades = new ArrayList<GradeDTO>();
+        for (Enrollment e : enrollments) {
+            Grade g = gradeRepository.findByEnrollmentIdAndAssignmentId(e.getEnrollmentId(), a.getAssignmentId());
+            Section s = a.getSection();
+            Course c = s.getCourse();
+            User u = e.getUser();
+
+            if (g != null) {
+                grades.add(new GradeDTO(g.getGradeId(), u.getName(), u.getEmail(),
+                        a.getTitle(), c.getCourseId(), s.getSecId(), g.getScore()));
+            } else {
+                grades.add(new GradeDTO(new Grade().getGradeId(), u.getName(), u.getEmail(),
+                        a.getTitle(), c.getCourseId(), s.getSecId(), null));
```

Comment:
I think you want to save the new grade here in the db to auto increment the grade id. See comment on L144.